### PR TITLE
[comments feature]: changes to 5 files to enable display of hard-coded comments

### PR DESCRIPTION
**Pull request by kiprot**
- Added gson dependency to pom.xml
- Added functions to DataServlet.java to convert arraylist to json
- Deleted defunct greetings button from index.html
- Added comments button to dinosaur.html

(photo1): before clicking comments button
Screenshot 2020-06-02 at 11 23 07 AM

(photo 2): showing hard coded comments
Screenshot 2020-06-02 at 11 23 22 AM

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -39,9 +39,24 @@ function toggleDetailsButton() {
   }
 }
 
-function getHelloMessage() {
-    fetch('/data').then(response => response.text()).then((message) => {
-        document.getElementById('message-container').innerText = message;
+// gets the data from server sides
```
Comment (octopus1000) [inline portfolio/src/main/webapp/script.js]: We usually starts comments with upper case.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -39,9 +39,24 @@ function toggleDetailsButton() {
...
+function getServerData() {
+    fetch('/data')  
+    .then(response => response.json()) 
+    .then((data) => {
+        const dataListElement = document.getElementById('comments');
+        dataListElement.innerHTML = '';
+        //for(var i = 0; i < data.length; i++) {
```
Comment (octopus1000) [inline portfolio/src/main/webapp/script.js]: Remove this?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -39,9 +39,24 @@ function toggleDetailsButton() {
...
-function getHelloMessage() {
-    fetch('/data').then(response => response.text()).then((message) => {
-        document.getElementById('message-container').innerText = message;
+// gets the data from server sides
+function getServerData() {
+    fetch('/data')  
+    .then(response => response.json()) 
+    .then((data) => {
+        const dataListElement = document.getElementById('comments');
+        dataListElement.innerHTML = '';
```
Comment (octopus1000) [inline portfolio/src/main/webapp/script.js]: Suggesting extracting logic of populating "comments" and merge it with createListElement to put visualization logic together.

**Review by octopus1000 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (kiprot) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (kiprot) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (kiprot) [reply portfolio/src/main/webapp/script.js]: alright. i made some changes.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -39,24 +39,22 @@ function toggleDetailsButton() {
   }
 }
 
-// gets the data from server sides
+// Gets the data from server sides
```
Comment (octopus1000) [inline portfolio/src/main/webapp/script.js:42]: Period at end of comment.

**Review by octopus1000 — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (kiprot) [reply portfolio/src/main/webapp/script.js:42]: okay

**Pull request merged**